src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/service/NotifyService.kt

```diff
@@ -20,10 +20,18 @@ import uk.gov.justice.digital.hmpps.hmppsinterventionsservice.exception.AsyncEve
 import uk.gov.justice.digital.hmpps.hmppsinterventionsservice.jpa.entity.AppointmentType
 import java.net.URI
 
-interface NotifyService {
+open class NotifyService {
   fun generateResourceUrl(baseURL: String, path: String, vararg args: Any): URI {
     return UriComponentsBuilder.fromHttpUrl(baseURL).path(path).buildAndExpand(*args).toUri()
   }
+
+  fun responsibleOfficerFirstName(responsibleOfficer: ResponsibleOfficer): String {
+    return responsibleOfficer.name?.split(' ')?.first() ?: ""
```

Comment:
This name is constructed during `ResponsibleOfficer(stringUtils.joinNullable(it.staff?.forenames, it.staff?.surname), ...)` and then split here -- can we pass down first/last names separately to avoid splitting after joining?

Reply:
we still need to split because the firstname can be any number of names.

i was trying my best to come up with the most sane representation of the name in the ResponsibleOfficer data class, regardless of how it's used. 

i have a habit of optimizing for stuff like this too but in reality string joining and splitting is the least of our worries when it comes to performance.

IMO

Reply:
I guess I'm worried about getting "surnames" in the mix -- if someone has `forenames="", surnames="Smith"`, addressing them as "Smith" might be awkward

Reply:
in your example case, would it be better to address them as `""` or `"smith"` though? at least it shows some reassurance they are the intended recipient of the email.